src/main/ApplicationUtils.cpp

```diff
@@ -195,13 +125,22 @@ setupApp(Config& cfg, VirtualClock& clock, uint32_t startAtLedger,
         return nullptr;
     }
 
-    // With in-memory mode, ledger state is not yet ready during this setup step
+    // With in-memory testing mode, ledger state is not yet ready during this
+    // setup step
     app->getLedgerManager().loadLastKnownLedger(
         /* restoreBucketlist */ false,
-        /* isLedgerStateReady */ !cfg.isInMemoryMode());
+#ifdef BUILD_TESTS
+        /* isLedgerStateReady */ !cfg.MODE_USES_IN_MEMORY_LEDGER
+#else
+        /* isLedgerStateReady */ true
+#endif
+    );
     auto lcl = app->getLedgerManager().getLastClosedLedgerHeader();
 
-    if (cfg.isInMemoryMode() &&
+    if (
+#ifdef BUILD_TESTS
```

Comment:
hm, this looks suspicious - in prod builds you don't want to be rebuilding state under any circumstances. setupApp shouldn't be called anywhere outside of tests, so probably needs to be behind BUILD_TESTS

Reply:
Good point, I'll harden this and remove the ledgerStateReady flag.

Reply:
sorry this is still not quite right: in prod paths we shouldn't be rebuilding state, ever. Right now you'd rebuild state at genesis, which I'm not sure would even work (I guess it does given that tests pass). I think this whole if statement should be behind BUILD_TESTS